domain/src/main/java/com/aconno/sensorics/domain/ifttt/outcome/InputToActionsResolverImpl.kt

```diff
@@ -7,29 +7,30 @@ import io.reactivex.disposables.Disposable
 class InputToActionsResolverImpl(
     actionsRepository: ActionsRepository
 ) : InputToActionsResolver {
-    private var disposables : MutableList<Disposable> = mutableListOf()
+    private var actionsProviderDisposable : Disposable? = null
 
     private var actionsMap : MutableMap<ResolverKey,MutableList<Action>> = mutableMapOf()
 
     init {
         actionsRepository.getAllActionsAsFlowable().subscribe {
-            actionsMap = actionsToMap(it)
-        }.also { disposables.add(it) }
+            updateMapWithActions(it)
+        }.also { actionsProviderDisposable = it }
```

Comment:
But if you use assignment statement then it looks like the main point of this statement is to assign value to actionsProviderDisposable, but it is not the main point - the main point is to subscribe to actions changes. So I think that this could be very misleading and that is why I have put this assignment into `also` method. What do you think about this, should I still convert this to the form that you proposed?

Reply:
That's a good point. I also don't like to assign the variable because it makes my first line too long and not clear.

My problem with your code is that it is also confusing. It looks like you are doing some extra action, but what you are actually do is to keep the reference for later disposal. Personally, this is worse than my first point.

Third solution which I often use is a pattern with an extension function. The name of the function is always `disposeOn<name-of-the-function-where-you-call-dispose>` in this case:
```
private val onFinalizeDisposables = CompositeDisposable()

private fun Disposable.disposeOnFinalize() {
    onFinalizeDisposables.add(this)
}

fun finalize() {
    ...
    onDinalizeDisposables.dispose()
    ...
}
```
or
```
private var onFinalizeDisposable: Disposable? = null

private fun Disposable.disposeOnFinalize() {
    onFinalizeDisposable = this
}

fun finalize() {
    ...
    onDinalizeDisposable?.dispose()
    ...
}
```

Then you use this in your code like:
```
someObservable.subscribe {
    ...
}.disposeOnFinalize()
```

You choose between Solution 1 and Solution 3.

Reply:
Ok, I choose Solution 1.